valhalla/baldr/graphconstants.h

```diff
@@ -739,6 +739,8 @@ enum class AccessType : uint8_t {
   kTimedAllowed = 6,
   kTimedDenied = 7,
   kDestinationAllowed = 8,
+  // TODO(nils): this is a problem. when evaluating restrictions in dynamiccost,
+  // we rely on the fact that non-time restrictions come first
   kMaxAxles = 9
```

Comment:
didn't think about it at the time: if a `maxaxle` restriction is combined with a `kTimeAllowed` restriction, then we won't respect the `maxaxle` one. this is because they are sorted in the graph tiles and maxaxles would be evaluated after timed restrictions which return true/false if active.

Reply:
but also there was no other way to do this without breaking the tile structure I guess. maybe we should just do 2 loops when evaluating restrictions, one for dimensions/axles etc, one for timed ones